src/auth.rs

```diff
@@ -17,8 +18,31 @@ pub enum Scopes {
 /// REST API authorization.
 #[derive(Clone, Debug, PartialEq)]
 pub struct Authorization {
+    /// Subject of the request.
     pub subject: String,
+
+    /// Authorization scopes available to the subject.
     pub scopes: Scopes,
+
+    /// The authentication mechanism that provided this authorization data.
+    ///
+    /// In cases where authentication is delegated to other microservices via
+    /// assertion headers, this field stores the original authentication
+    /// mechanism that initially authenticated the subject.
+    pub auth_type: String,
+
+    /// Issuer of this request.
+    ///
+    /// When a system is operating on behalf of a subject, the subject field
+    /// contains the subject of the request, while the issuer field contains
+    /// the system that issued the request.
+    pub issuer: Option<String>,
+
+    /// Expiry deadline for this authorization data.
+    ///
+    /// This is used when the authorization data is cached, used to start a
+    /// session, or is used to construct a token passed back to the client.
+    pub expiry_deadline: Option<DateTime<FixedOffset>>,
```

Comment:
This should be UTC; all times in new MSw code should be in UTC unless there is a very good reason otherwise (see the FSpec / HLD template). I don't think keeping the timezone of the deadline around is very useful, and it could cause confusion when compared with other (UTC) timestamps.